client/src/components/AddGame.jsx

```diff
@@ -0,0 +1,221 @@
+          </div>
+
+          <div className="form-group col-md-6">
+            <label htmlFor="publisher">Publisher</label>
+            <input type="text" className="form-control" name="publisher" onChange={this.handleChange} />
+          </div>
+        </div>
+        
+        <div className="form-group">
+          <label htmlFor="categories">Categories</label>
+          <Select isMulti options={this.props.category_list} className="basic-multi-select" classNamePrefix="select" name="categories" onChange={this.handleCategoryChange.bind(this)} />
+        </div>
+        <h3 className="mt-4">Game Details</h3>
+
+        <div className="form-row">
+          <div className="form-group col-md-6">
+            <label>Players</label>
+            <div className="d-flex flex-row">
+              <input type="number" className="form-control" name="min_players" placeholder="min" onChange={this.handleChange} />
+              <div className="mx-2 mt-1"> &ndash; </div>
+              <input type="number" className="form-control" name="max_players" placeholder="max" onChange={this.handleChange} />
+            </div>
+          </div>
+
+
+          <div className="form-group col-md-6">
+            <label>Playtime (minutes)</label>
+            <div className="d-flex flex-row">
+              <input type="number" className="form-control" name="min_playtime" placeholder="min" onChange={this.handleChange} />
+              <div className="mx-2 mt-1"> &ndash; </div>
+              <input type="number" className="form-control" name="max_playtime" placeholder="max" onChange={this.handleChange} />
+            </div>
+          </div>
+        </div>
+
+        <div className="form-row">
+          <div className="form-group col-md-6">
+            <label htmlFor="year">Year</label>
+            <input type="number" className="form-control" name="year" onChange={this.handleChange} />
+          </div>
+
+          <div className="form-group col-md-6">
+            <label htmlFor="age">Minimum Age</label>
+            <input type="number" className="form-control" name="min_age" onChange={this.handleChange} />
+          </div>
+        </div>
+
+        <input type="submit" className="btn btn-lg btn-primary" value="Save Game" />
+
+      </form>
+    );
+  }
+
+  handleChange(event) {
+    const name = event.target.name;
+    const value = name === 'image' ? event.target.files[0] : event.target.value;
+
+    this.setState({ [name]: value });
+
+    if (name === 'name') {
+      // TODO: Check database for similar game names
+    }
+  }
+
+  handleCategoryChange(categories) {
+    this.setState({ categories: categories });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    let values = {...this.state};
+    if (values.categories) {
+      values.categories = values.categories.map(e => e.value);
+    }
+
+    // Upload image file to S3
+    if (values.image) {
+      let image_data = {
```

Comment:
Nit: I'd prefer it be called imageData here, just because JavaScript uses camelCase and not snake_case to name their variables.

Reply:
Variable name has been updated. I also updated the variables for the form field names to use camel case as well